src/Components/FilterControls.js

```diff
@@ -1,86 +1,79 @@
 import React, { Component } from 'react';
 import CardContainer from './CardContainer';
+import '../FilterControls.css';
 
 class FilterControls extends Component {
   constructor(props) {
     super(props);
     this.state = {
       location: '',
-      time: '',
+      dayOrNight: '',
       date: ''
     }
   }
 
-  getClickedLocation = (event) => {
-    let selectedLocation = event.target.value;
+  getClickedValue = (event) => {
     this.setState({
-      location: selectedLocation
-    }, this.filterByLocation)
+      [event.target.id]: event.target.value
+    }, this.filterParades)
   }
 
-  
-  filterByLocation = () => {
-    return this.props.parades.filter(parade => {
-      return parade.location === this.state.location
-    })
-  }
-  
-  getClickedTime = (event) => {
-    let selectedTime = event.target.value;
-    this.setState({
-      time: selectedTime
-    }, this.filterByTime)
-  }
-  
-  filterByTime = () => {
-    return this.props.parades.filter(parade => {
-      return parade.dayOrNight === this.state.time
+  filterParades = () => { 
+    let keyArr = ['location', 'dayOrNight', 'date']
+    let filteredParades = this.props.parades.filter(parade => {
+      return keyArr.every(key => {
+        return this.state[key] === parade[key] || this.state[key] === ''
+      })
     })
+    return filteredParades
```

Comment:
Ahh, much better refactored version of the filtering - disregard feedback around this on your last PR.